docs/guide/creating_a_gem.md

```diff
@@ -0,0 +1,148 @@
+---
+layout: default
+title: Creating a Gem
+---
+
+# Creating a Gem
+
+Start by creating a new Rails engine with the following arguments:
+
+```sh
+rails plugin new shared-view_components \
+  --full \
+  --skip-namespace \
+  --skip-action-mailer \
+  --skip-action-mailbox \
+  --skip-action-text \
+  --skip-active-record \
+  --skip-active-job \
+  --skip-active-storage \
+  --skip-action-cable \
+  --skip-turbolinks \
+  --skip-jbuilder \
+  --skip-gemfile-entry # only use this when adding a gem into an existing Rails app
+```
+
+Update the gemspec. Add the dependency:
+
+```ruby
+spec.add_dependency "view_component", ">= 2.32"
+```
+
+Afterwards, add the require command into `shared/view_components.rb`:
+
+```ruby
+require "view_component/engine"
+```
+
+Let's create our first component:
+
+```sh
+✗ bin/rails g component Shared::Example title --skip-namespace
+      create  app/components/shared/example_component.rb
+      invoke  test_unit
+      create    test/components/shared/example_component_test.rb
+      invoke  erb
+      create    app/components/shared/example_component.html.erb
+```
+
+Update `example_component.html.erb` to the following:
+
+```erb
+<span title="<%= @title %>"><%= content %></span>
+```
+
+Now, let't write a test for the component. Open `test/components/example_component_test.rb`
+
+```ruby
+require "test_helper"
+
+class ExampleComponentTest < ViewComponent::TestCase
+  def test_renders
+    assert_equal(
+      %(<span title="Hello">World!</span>),
+      render_inline(Shared::ExampleComponent.new(title: "Hello").with_content("World!")).css("span").to_html
+    )
+  end
+end
+```
+
+Run the test:
+
+```sh
+✗ bin/rails test
+Run options: --seed 33407
+
+# Running:
+
+..
+
+Finished in 0.047348s, 42.2404 runs/s, 42.2404 assertions/s.
+2 runs, 2 assertions, 0 failures, 0 errors, 0 skips
+```
+
+Great! Now lets add test the gem in the dummy test app. To start, we need to add `require "view_component/test_helpers"` to `test_helper.rb`
```

Comment:
I think this line is not grammatically correct.

Also, might we use a term besides `dummy`? Some might find it less than inclusive ❤️

Reply:
We went for `sandbox` in https://github.com/github/view_component/pull/985#issuecomment-870604847

Reply:
I just noticed that the `dummy` name comes from the `test/dummy` folder created by the `rails plugin new` generator. Changing the term here would mean that this folder has to be renamed by the user after running `rails plugin new`. Not very practical, I guess. Maybe you should advocate for changing the term in Rails?